File: 2025_03/leetcode2025-03-18.py
from typing import List

class Solution:
    def diagonalPrime(self, nums: List[List[int]]) -> int:
        def isPrime(x):
            if x==1: return False
            for i in range(2, int(x**(0.5))+1):
                if x%i == 0:
                    return False
            return True
        myMax = lambda x,y : x if x>y else y
        res = 0
        n = len(nums)
        for i in range(n):
            if isPrime(nums[i][i]):
                res = myMax(res, nums[i][i])
            if isPrime(nums[i][n-i-1]):
                res = myMax(res, nums[i][n-i-1])
        return res

if __name__ == "__main__":
    s = Solution()
    nums = [[1,2,3],[5,6,7],[9,10,11]]
    print(s.diagonalPrime(nums))

File: 2025_03/leetcode2025-03-18.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
private:
    bool isPrime(int x){
        if (x==1) return false;
        for (int i=2; i*i<=x; i++){
            if (x%i == 0){
                return false;
            }
        }
        return true;
    }
public:
    int diagonalPrime(vector<vector<int>>& nums) {
        int n = nums.size();
        int res = 0;
        for (int i=0; i<n; i++){
            if (isPrime(nums[i][i])){
                res = max(res, nums[i][i]);
            }
            if (isPrime(nums[i][n-i-1])){
                res = max(res, nums[i][n-i-1]);
            }
        }
        return res;
    }
};

int main(){
    Solution s;
    vector<vector<int>> nums = {{1,2,3},{5,6,7},{9,10,11}};
    cout << s.diagonalPrime(nums) << endl;
}
